qiskit/algorithms/amplitude_amplifiers/amplitude_amplifier.py

```diff
@@ -46,6 +46,7 @@ def __init__(self) -> None:
         self._top_measurement = None
         self._assignment = None
         self._oracle_evaluation = None
+        self._circuit_results = None
```

Comment:
Can we add `self._max_probability` here too. I see that is a getter/setter below but we don't setup the instance var here